src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs

```diff
@@ -0,0 +1,190 @@
+    {
+        private readonly string displayName;
+        internal const int EntireRepositoryFilterHandlerFilterId = 22;
+
+        public EntireOragnizationFilterHandler()
+        {
+            this.displayName = "Entire Organization";
+        }
+
+        public override int FilterId => EntireRepositoryFilterHandlerFilterId;
+
+        public override string FilterDisplayName => this.displayName;
+
+        public override IEntryFilter GetFilter(out string displayText)
+        {
+            displayText = this.displayName;
+            return new ItemOriginFilter(ItemOrigin.IndexedInOrganization);
+        }
+
+        protected override void OnActivated()
+        {
+            base.OnActivated();
+            // User just selected the broadest scope. See if we need to re-do the search
+        }
+    }
+
+    [Export(typeof(IScopeFilterFactory))]
+    [TableManagerIdentifier("FindAllReferences*")]
+    [TableManagerIdentifier("FindResults*")]
+    [DeferCreation(OptionName = RichNavOptions.RichNavAvailableOptionName)] // This factory will not be loaded unless this option is set to Boolean true
+    [Name(PredefinedScopeFilterNames.EntireRepositoryScopeFilter)]
+    [Order(Before = PredefinedScopeFilterNames.LoadedSolutionScopeFilter)]
+    internal class EntireRepositoryFilterHandlerFactory : IScopeFilterFactory
+    {
+        [ImportingConstructor]
+        public EntireRepositoryFilterHandlerFactory()
+        {
+        }
+
+        public IErrorListFilterHandler CreateFilter(IWpfTableControl tableControl)
+        {
+            return new EntireRepositoryFilterHandler();
+        }
+    }
+
+    internal class EntireRepositoryFilterHandler : FilterHandlerBase
+    {
+        private readonly string displayName;
+        internal const int EntireRepositoryFilterHandlerFilterId = 21;
+
+        public EntireRepositoryFilterHandler()
+        {
+            this.displayName = "Entire Repository";
+        }
+
+        public override int FilterId => EntireRepositoryFilterHandlerFilterId;
+
+        public override string FilterDisplayName => this.displayName;
+
+        public override IEntryFilter GetFilter(out string displayText)
+        {
+            displayText = this.displayName;
+            return new ItemOriginFilter(ItemOrigin.IndexedInRepo);
+        }
+
+        protected override void OnActivated()
+        {
+            base.OnActivated();
+            // User just selected a broader scope. See if we need to re-do the search
+        }
+    }
+
+    internal class ItemOriginFilter : IEntryFilter
+    {
+        private readonly ItemOrigin targetOrigin;
+
+        internal ItemOriginFilter(ItemOrigin targetOrigin)
+        {
+            this.targetOrigin = targetOrigin;
+        }
+
+        public bool Match(ITableEntryHandle entry)
+        {
+            Requires.NotNull(entry, nameof(entry));
+
+            if (entry.TryGetValue(StandardTableKeyNames.ItemOrigin, out ItemOrigin entryOrigin))
```

Comment:
so this works by grabbing an ITemOrigin out of hte entry?  i assume this means i should be adding that ItemOrigin to the entry based on the values i get from richnav?    Is there a set of well-known origins?

Reply:
also, i don't think rich nav provides this info.  should i just assume all the results i get back for searching a particular scope *are* in that scope?

Reply:
RichNav is providing this info. Note that when searching in Organization scope, you might be also getting results specific from your Repository. They would have a different scope

Reply:
The key names are in the `Microsoft.VisualStudio.Shell.Framework` dll, in `Microsoft.VisualStudio.Shell.TableManager.StandardTableKeyNames` type
![image](https://user-images.githubusercontent.com/1673956/78167606-9a429480-7403-11ea-9434-409c589a4852.png)

Reply:
> RichNav is providing this info.

Do you know where?  All they pass us back is a LocationWithText.  Does that have Origin data in it?